concurrent_test.go

```diff
@@ -0,0 +1,98 @@
+package bbolt_test
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	bolt "go.etcd.io/bbolt"
+	"go.etcd.io/bbolt/internal/btesting"
+)
+
+func TestRepeatableRead(t *testing.T) {
+	const (
+		readInterval  = 50 * time.Millisecond
+		writeInterval = 10 * time.Millisecond
+		testDuration  = 30 * time.Second
+	)
+
+	bucket := []byte("data")
+	key := []byte("foo")
+
+	// prepare the db
+	t.Log("Create the db")
+	db := btesting.MustCreateDB(t)
+	err := db.Update(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucket(bucket)
+		return err
+	})
+	require.NoError(t, err)
+
+	// start readonly transactions
+	stopCh := make(chan struct{}, 1)
+	var wg sync.WaitGroup
+	t.Log("Create readonly transactions")
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-stopCh:
+				t.Log("Readonly transaction finished.")
+				return
+			default:
+			}
+
+			err := db.View(func(tx *bolt.Tx) error {
+				b := tx.Bucket(bucket)
+				initialVal := b.Get(key)
+				for i := 0; i < 10; i++ {
+					val := b.Get(key)
+					require.Equal(t, initialVal, val)
+					time.Sleep(readInterval)
+				}
+
+				return nil
+			})
+			require.NoError(t, err)
+		}
+	}()
+
+	// start write transactions
+	t.Log("Create write transactions")
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-stopCh:
+				t.Log("Write transaction finished.")
+				return
+			default:
+			}
+
+			err := db.Update(func(tx *bolt.Tx) error {
+				b := tx.Bucket(bucket)
+				v := fmt.Sprintf("%04d", rand.Intn(10000))
+				putErr := b.Put(key, []byte(v))
+				require.NoError(t, putErr)
+
+				return nil
+			})
+			require.NoError(t, err)
```

Comment:
I assume require.... is likely to call (https://pkg.go.dev/testing#T.FailNow), that shouldn't be called from other goroutines than the main tests go-routine. 

Please consider `errgroup` instead of `wg`
https://pkg.go.dev/golang.org/x/sync/errgroup